sdk/servicebus/service-bus/src/core/streamingReceiver.ts

```diff
@@ -554,6 +568,8 @@ export class StreamingReceiver extends MessageReceiver {
     }
 
     this._isDetaching = true;
+    this.isReceivingMessages = false;
```

Comment:
This flag is used externally to determine whether or not a `receiveMessages()` or `subscribe()` call should throw the "already receiving" error or let the call succeed.

At this point, we are attempting to recover a broken streaming receiver. Setting `this.isReceivingMessages` to false here would allow the user to call `subscribe()` again while this recovery is going on. That would not be desirable right?

Reply:
I removed it - it was redundant anyways.

Ultimately it should be okay either way because their call will run into (and block) on the same internal init() call at some point so it'll essentially put everyone in lockstep.